### ISSUE TITLE
CFTime vs. pd.datetime

### ISSUE THREAD

**Issue by ks905383** [xgcm/xrft#98]
Found out the type error issue. 

The dataset I was using has time imported `cftime` format (since it's a `noleap` calendar, which are imported as `cftime.datetime` index per `xarray` docs). 

Unfortunately, `cftime` formatted dates cause an error when `dft` is run on the `time` dimension, in the `# verify even spacing of input coordinates` section:

```
for d in dim:
        coord = da[d]
        diff = np.diff(coord)
        if pd.api.types.is_timedelta64_dtype(diff):
            # convert to seconds so we get hertz
            diff = diff.astype('timedelta64[s]').astype('f8')
        delta = diff[0]
        if not np.allclose(diff, diff[0], rtol=spacing_tol):
            raise ValueError("Can't take Fourier transform because "
                             "coodinate %s is not evenly spaced" % d)
        delta_x.append(delta)
```

Specifically, the `if` statement only converts `timedelta64` type values to integers; in my case, `diff` is kept as `datetime.timedelta`. Since `np.allclose()` requires numeric inputs, the code fails at `if not np.allclose(diff,diff[0],rtol=spacing_tol)`. 

**Dealing more generally with `noleap` calendars**
A related issue is that even converting `diff` to a supported format (using `ds.indexes['time'].to_datetimeindex()` poses a problem - 

```
/home/kschwarzwald/.conda/envs/climate1/lib/python3.7/site-packages/ipykernel_launcher.py:1: RuntimeWarning: Converting a CFTimeIndex with dates from a non-standard calendar, 'noleap', to a pandas.DatetimeIndex, which uses dates from the standard calendar.  This may lead to subtle errors in operations that depend on the length of time between dates.
  """Entry point for launching an IPython kernel.
```
In other words, `diff` will now have twice the value on "leap" days, which don't exist in the original dataset, and therefore trigger the "coordinate is not evenly spaced error" (and presumably cause issues with the Fourier transform.

**Proposed Solution** 
A first useful addition would probably be raising a `TypeError` on `cftime` dimensions. 

But further, I think this can be fixed by just adding a
